akka-http-core/src/main/resources/reference.conf

```diff
@@ -374,6 +378,10 @@ akka.http {
     // FIXME: unify with server-side part (by importing or similar to parsing)
     http2 {
       # The maximum number of request per connection concurrently dispatched to the request handler.
+      # This limit is enforced as soon as the connection between the peers is established. Enforcing
+      # the limit even before the SETTINGS/SETTINGS_ACK exchange has completed, means
+      # that we will refuse extra streams/push promises that were sent by the client right after 
+      # the connection was established but before it received our SETTINGS.
```

Comment:
As discussed, `"infinite"` is not supported and the value of `max-concurrent-streams` will also be used as the `parallelism` on the stream.